testsuites/CBLTester/CBL_Functional_tests/APITests/test_collection.py

```diff
@@ -0,0 +1,264 @@
+        random_string(1),
+        random_string(6),
+        random_string(128),
+        "_{}".format(random_string(6)),
+        "{}_".format(random_string(6)),
+        "_{}_".format(random_string(6)),
+        random_string(6, digit=True),
+        random_string(6).upper(),
+    ])
+    # Test to check default collection name in CBL
+    def test_defaultCollection(self, db_name):
+        db = self.db_obj.create(db_name)
+        collection = self.db_obj.defaultCollection(db)
+        collectionName = self.collection_obj.collectionName(collection)
+        expected_name = '_default'
+        assert collectionName == expected_name, "Wrong Default Collection Name"
+
+    @pytest.mark.parametrize("db_name, collectionName", [
+        (random_string(6), random_string(6)),
+        (random_string(6), random_string(6)),
+        (random_string(6), random_string(6)),
+        (random_string(6), random_string(6)),
+        (random_string(6), random_string(6)),
+        (random_string(6), random_string(6)),
+    ])
+    def test_createCollection(self, db_name, collectionName):
+        db = self.db_obj.create(db_name)
+        scopeName = '_default'
+        created_collection = self.db_obj.createCollection(db, collectionName, scopeName)
+        created_collection_name = self.collection_obj.collectionName(created_collection)
+        assert created_collection_name == collectionName, "Collection name doesn't match"
+
+    @pytest.mark.parametrize("db_name, collectionName, no_of_document", [
+        (random_string(6), random_string(6), 9),
+        (random_string(6), random_string(6), 99),
+        (random_string(6), random_string(6), 999),
+    ])
+    def test_documentCount(self, db_name, collectionName, no_of_document):
+        db = self.db_obj.create(db_name)
+        scopeName = '_default'
+        created_collection = self.db_obj.createCollection(db, collectionName, scopeName)
+        for i in range(0, no_of_document):
+            doc = self.doc_obj.create()
+            self.collection_obj.saveDocument(created_collection, doc)
+        document_count = self.collection_obj.documentCount(created_collection)
+        assert document_count == no_of_document, "Document count doesn't match"
+
+    # Test to delete collection.
+    @pytest.mark.parametrize("db_name, collectionName", [
+        (random_string(6), random_string(6)),
+        (random_string(6), random_string(6)),
+        (random_string(6), random_string(6)),
+        (random_string(6), random_string(6)),
+    ])
+    def test_deleteCollection(self, db_name, collectionName):
+        db = self.db_obj.create(db_name)
+        scopeName = "_default"
+        self.db_obj.createCollection(db, collectionName, scopeName)
+        collectionNames = self.db_obj.collectionsInScope(db, scopeName)
+        self.db_obj.deleteCollection(db, collectionName, scopeName)
+        collectionNames2 = self.db_obj.collectionsInScope(db, scopeName)
+        deleted_item = []
```

Comment:
For, simplicity, is it possible to check if the collection name exists after it was deleted?